src/pandroid/app/src/main/java/com/panda3ds/pandroid/PandaGlRenderer.java

```diff
@@ -0,0 +1,79 @@
+package com.panda3ds.pandroid;
+
+import javax.microedition.khronos.egl.EGLConfig;
+import javax.microedition.khronos.opengles.GL10;
+
+import static android.opengl.GLES32.*;
+
+import android.content.res.Resources;
+import android.opengl.GLSurfaceView;
+import android.util.DisplayMetrics;
+import android.util.Log;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+
+public class PandaGlRenderer implements GLSurfaceView.Renderer {
+    int screenWidth, screenHeight;
+    int screenTexture;
+    public int screenFbo;
+
+    PandaGlRenderer() {
+        super();
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        if (screenTexture != 0) {
+            glDeleteTextures(1, new int[]{screenTexture}, 0);
```

Comment:
Java is a meme language I can't even

Is this how it's actually meant to be done in Java

Reply:
yes afaik this is the equivalent of a destructor and the java<->Gl interaction needs you to pass the arguments like that afaics